doc/src/guides/solving/solve-polynomial.md

```diff
@@ -0,0 +1,514 @@
+
+## Example of Finding the Roots of a Polynomial Algebraically
+
+Here is an example of solving a polynomial algebraically:
+
+```py
+>>> from sympy import roots
+>>> from sympy.abc import x, a, b, c
+>>> roots(a*x**2 + b*x + c, x)
+{-b/(2*a) - sqrt(-4*a*c + b**2)/(2*a): 1,
+ -b/(2*a) + sqrt(-4*a*c + b**2)/(2*a): 1}
+```
+
+This example reproduces the [quadratic
+formula](https://en.wikipedia.org/wiki/Quadratic_formula).
+
+## Functions to Find the Roots of a Polynomial
+
+There are several functions that you can use to find the roots of a polynomial:
+- {func}`~.solve` is a general solving function which can find roots, though is
+  less efficient than {meth}`~sympy.polys.polytools.Poly.all_roots` and is the
+  only function in this list that does not convey the multiplicity of roots
+- {func}`~.roots` computes the symbolic roots of a univariate polynomial; will
+fail for most high-degree polynomials (five or greater)
+- {func}`~.nroots` computes numerical approximations of the roots of any
+polynomial whose coefficients can be numerically evaluated
+- {func}`~sympy.polys.rootoftools.RootOf` can find all the roots exactly of a
+polynomial of arbitrarily large degree
+- {func}`~.real_roots` can find all the real roots exactly of a polynomial of
+arbitrarily large degree; because it finds only the real roots, it can be more
+  efficient than functions that find all roots
+- {meth}`~sympy.polys.polytools.Poly.all_roots` can find all the roots exactly
+of a polynomial of arbitrarily large degree
+- {func}`~.factor` factors a polynomial into irreducibles and can reveal that
+  roots lie in the coefficient ring
+
+Each will be used on this page.
+
+## Guidance
+
+Refer to
+[](solving-guidance.md#include-the-variable-to-be-solved-for-in-the-function-call)
+and [](solving-guidance.md#use-exact-values).
+
+## Solve (Find the Roots of) a Polynomial
+
+You can solve a polynomial algebraically in several ways. The one to use depends
+on whether you
+- want an algebraic or numeric answer
+- want the multiplicity of each root (how many times each root is a solution).
+  In the `expression` below representing $(x+2)^2(x-3)$, the root -2 has a
+  multiplicity of two because $x+2$ is squared, whereas 3 has a multiplicity of
+  one because $x-3$ has no exponent. Similarly, for the `symbolic` expression,
+  the root $-a$ has a multiplicity of two and the root $b$ has a multiplicity of
+  one.
+
+```py
+>>> from sympy import solve, roots, real_roots, factor, nroots, RootOf, expand
+>>> from sympy import Poly
+>>> expression = (x+2)**2 * (x-3)
+>>> symbolic = (x+a)**2 * (x-b)
+```
+
+### Algebraic Solution Without Root Multiplicities
+
+You can use SymPy's standard {func}`~.solve` function, though it will not return
+the multiplicity of roots:
+
+```py
+>>> solve(expression, x, dict=True)
+[{x: -2}, {x: 3}]
+>>> solve(symbolic, x, dict=True)
+[{x: -a}, {x: b}]
+```
+
+Refer to [](solve-equation-algebraically.md) for more about using
+{func}`~.solve`.
```

Comment:
I think it's worth clarifying here the relationship between `solve` and the other functions. Basically `solve` will first try using `roots` and if that doesn't work it will try using `all_roots`. For cubics and quartics that means that it will use radical formulae from `roots` rather than `RootOf` even if `RootOf` is possible. However `solve` has two parameters `cubics` and `quartics` that can control this:
```python
In [4]: solve(x**4 - x + 1, quartics=False)
Out[4]: 
⎡       ⎛ 4           ⎞         ⎛ 4           ⎞         ⎛ 4           ⎞         ⎛ 4           ⎞⎤
⎣CRootOf⎝x  - x + 1, 0⎠, CRootOf⎝x  - x + 1, 1⎠, CRootOf⎝x  - x + 1, 2⎠, CRootOf⎝x  - x + 1, 3⎠⎦
```